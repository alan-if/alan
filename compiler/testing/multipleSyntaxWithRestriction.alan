Syntax give_to = give (o) 'to' (a)
    Where a Isa actor Else "Only actors will appreciate gifts."

Syntax give_to = give (a) (o)
    Where a Isa actor Else "Only actors will appreciate gifts."

Add To object
  Verb give_to
    When o Does
	"Give" Say a. Say o. "."
	"Give" Say o. "to" Say a. "."
  End Verb.
End Add.

Add To actor
  Verb give_to
    When a Does
	"Give" Say a. Say o. "."
	"Give" Say o. "to" Say a. "."
  End Verb.
End Add.

The l Isa location
End THe l.

The ball Isa object At l
End THe ball.

The policeman Isa actor At l
End The policeman.

Start At l.


