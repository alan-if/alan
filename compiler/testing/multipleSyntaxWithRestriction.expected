########## multipleSyntaxWithRestriction ##########


multipleSyntaxWithRestriction.alan

    1.  Syntax give_to = give (o) 'to' (a)
=====>         1

  *1*   205 W : This is the first occurrence of syntax for 'give_to', see
                errors below.

    5.      Where a Isa actor Else "Only actors will appreciate gifts."
=====>      1

  *1*   250 E : You can only declare restrictions for the first syntax for verb
                'give_to'. The restrictions for subsequent syntaxes must be
                empty and will use the same as for the first.


        1 error(s).
        1 warning(s).
        1 informational message(s).

